### ISSUE TITLE
Create search/filter box component

### ISSUE THREAD

**Issue by hdrury1** [elifesciences/elife-xpub#347]
**User Story:**

_As a submitter I want to filter the list of editors in order to narrow down the number I have to choose from._

**Product requirements/Acceptance criteria:**

Functional:

- The search box should be propulated by default with the MSA(s) already selected on page 3 of the submission wizard.
- An entry in the search box should filter down the number of possible selections to choose from. (see ticket)
- A user can filter by MSA, keyword or name. (see  #349)
- Clicking on the search bar enables the user to enter input.
- When a user begins to enter input into the search bar, a dropdown appears with a list of suggested terms based on input. Clicking on one of these suggestions populates the filter box. (see ticket)
- Clicking the search icon or hitting the enter key updates the list of people to reflect the search term(s) entered into the search box.

**Relevant links to design(s):**

https://invis.io/9HLUEH9AXRF#/305823519_picker-Author
https://projects.invisionapp.com/share/FUMJS349R2E#/screens/305823534_staff-Picker-Reviewers-Search

**Relevant files**
[search_icon.zip](https://github.com/elifesciences/elife-xpub/files/2223272/search_icon.zip)

**Comment by dryhten** [comment]
@hdrury1 I have a few questions about this

1. When you click on the search box, do you get a dropdown menu with all the options?
2. When you start typing something, you will get a dropdown list with all the options that match the string you typed. Should this match only happen at the beginning of the string or can it happen anywhere? For example, if I start typing "Mike", do I get all the people who have a first name of Mike or can I can the ones that have a last name of "Mike" too?
3. Do I understand correctly that we can search by name, institution and subject area all in the same search box? If that's the case, is it fine to show all results that matched no matter what field matched? For example, if I search for Max, I would expect to see all the people called Max and also people from the Max Planck Institute.
4. Is anyone supposed to be able to add a new reviewer to the system using `Add new +`? Some people might add gibberish and showing this to other submitters will give a bad experience. Or do we plan to not add the new reviewers entered manually to the global scope? (e.g. save them for other users).
5. As far as I remember, the list of pods won't change as you start typing something, is that correct?

**Comment by dryhten** [comment]
I have another question for this, are we supposed to get the list of people (e.g. reviewers) from an elife api or do we store it in the db ourselves? The same goes for MSA (I assume this is Major Subject Area?) and institutions.

Also, I assume the list of available options that shows up as you type should be done as part of this other issue https://github.com/elifesciences/elife-xpub/issues/349.

Is `Add new +` text/button described anywhere or do we just skip it for now?

**Comment by hdrury1** [comment]
Hi @dryhten, to answer your queries:

1. No. Suggestions are offered based on input, so no suggestion is offered until the first character is typed.

2. Yes, match can happen anywhere.

3. Correct. We may need to refine this logic based on testing, but for now let's just show everything.

4. Only staff and editors will be able to `Add New +`, so we don't need to worry about this functionality until later in development (not needed for wizard).

5. Correct. Pods update on hitting enter or the search icon.

6. Data should be coming from eLife API (`/people`) which has everything we need for this stage of the process as we're only looking for editors.

7. Yes... I think it makes sense to do this work as part of a separate issue.

8. See above. We can skip for now.

**Comment by dryhten** [comment]
As discussed on the standup, the selected options (the blue ones at the bottom) are not part of this search box. The styles have been reverted in https://github.com/elifesciences/elife-xpub/pull/407/commits/34e6e4978ded87731797ec6ae9c264b95d0a3f8e

The search icon uploaded here replaced the material one in https://github.com/elifesciences/elife-xpub/pull/407/commits/66586ba2098d887094e1a933afa8df17990fb62c

The only thing left here is remove background color for selected items. Once we select an item from the dropdown list in the search box, what's the expected behavior when we start typing again? @hdrury1 
Is it similar to the subject area dropdown, when the selected item forms an entity you can cancel by clicking an x?

**Comment by hdrury1** [comment]
Hey @dryhten. This search box is a completely different component from the subject area dropdown. In this instance, the blue tags refer to selected persons rather than selected search terms.

Functionality should be the same as Google suggestions- once you have selected an item from the dropdown, continuing to type should further refine the list of sugestions.

**Comment by hdrury1** [comment]
Hey @dryhten. This search box is a completely different component from the subject area dropdown. In this instance, the blue tags refer to selected persons rather than selected search terms.

Functionality should be the same as Google suggestions- once you have selected an item from the dropdown, continuing to type should further refine the list of suggestions.

**Comment by dryhten** [comment]
Ok cool. As mentioned in the slack conversation, I went with [react-autosuggest](https://react-autosuggest.js.org/) since it seems like a good fit for what we need to do.

What's left to be done

- [ ] Integrate search box at page level. Right the component accepts a list of suggestions from a higher component (e.g. the page). As far as I remember this was the outcome of my discussion with @tamlyn, that the eLife api should be queried at the page level and results should be cached.
 - [ ] Wait for detailed designs for the search box. Right now I styled it based on the TextField component.
 - [ ] Should the text you enter go under the search icon as you type? Right now it does, I assume this is not the intended behavior so it should be fixed.
 - [ ] Switch from CSS modules to styled components. React autosuggest has built-in support for CSS modules but not for styled components. Styling the search box with styled components might be tricky and deserves a separate issue. We would need that for consistency (we use styled components everywhere else) and to be able to access theme variables - I couldn't find a way to do this with CSS modules so everything is hardcoded in at the moment.
